xarray/core/merge.py

```diff
@@ -123,16 +124,24 @@ def unique_variable(
         combine_method = "fillna"
 
     if equals is None:
-        out = out.compute()
+        # first check without comparing values i.e. no computes
         for var in variables[1:]:
-            equals = getattr(out, compat)(var)
+            equals = getattr(out, compat)(var, equiv=lazy_array_equiv)
             if not equals:
```

Comment:
Should this be `if equals is not None`?

Reply:
I think this is right (as above) but the next one is wrong. I've fixed that and added a test.